java/arcs/core/storage/BackingStore.kt

```diff
@@ -53,7 +53,21 @@ class BackingStore<Data : CrdtData, Op : CrdtOperation, T>(
      */
     suspend fun getLocalData(referenceId: String) = store(referenceId).store.getLocalData()
 
-    suspend fun clearStoresCache() = storeMutex.withLock { stores.clear() }
+    /** Removes [DirectStore] caches and closes those that can be closed safely. */
+    fun clearStoresCache() {
+        while (!storeMutex.tryLock()) { /* Wait */ }
+        try {
+            stores.forEach { _, sr ->
+                /** Aligns with the logic of [stores]'s [LruCacheMap.onEvict]. */
+                if (!sr.store.closed) {
+                    sr.store.close()
+                }
```

Comment:
can we extract this block to a function eg closeStore and use it both here and as onEvict?

Reply:
Yes, done!